.emacs.d/lisp/code/languages/csharp.el

```diff
@@ -0,0 +1,91 @@
+;;  -*- lexical-binding: t; -*-
+
+;;; The following keybindings are available when editing C# files using omnisharp:
+;; C-RET: Do a refactoring at point. Prompt for which refactoring in the minibuffer.
+;; M-RET: Same as C-RET
+;; C-c o r: Intelligently rename the symbol at point, throughout project
+;; M-.: Jump to definition of symbol at point
+;; C-.: Find usages of symbol at point
+;; C->: Find implementations of symbol at point
+;; C-M-\\: Reformat the region
+
+
+(defun kotct/omnisharp-install-server ()
+  "Install omnisharp-roslyn, or, if using Arch Linux, make sure it is installed
+and then install the correct wrapper."
+  (interactive)
+  (let ((os (shell-command-to-string "lsb_release -ds")))
+    (if (string= os "\"Arch Linux\"\n")
+        (if (= 0 (call-process-shell-command "pacman -Q msbuild-stable omnisharp-roslyn"))
+            (progn
+              (customize-save-variable 'omnisharp-server-executable-path "~/.emacs.d/bin/omnisharp")
+              (message "Ready to roll! Do M-x omnisharp-mode to start OmniSharp."))
+          (message "Looks like you're using Arch Linux!
+Please insall omnisharp-roslyn and msbuild-stable from the AUR.
+When you're done, try running M-x kotct/omnisharp-install-server again."))
+      ;; default to built-in installation
+      (omnisharp-install-server nil))))
+
+(defun kotct/omnisharp-maybe-start ()
+  "Start omnisharp, or prompt user to install omnisharp-roslyn."
+  (interactive)
+  (require 'omnisharp)
+  ;; this is approximately the check omnisharp does to find the server
+  (let ((path (or omnisharp-server-executable-path
+                  (omnisharp--server-installation-path t))))
+    (if (f-exists-p path)
+        (omnisharp-mode +1)
+      (message "Heads up! For sick autocompletion and autodocs, you should install OmniSharp!
+Do M-x kotct/omnisharp-install-server and we'll take care of it."))))
+
+(add-hook 'csharp-mode-hook (lambda ()
```

Comment:
I forget, is our coding style to put `lambda` openings on the next line, indented to the same level as the symbol preceeding?

Reply:
I don't think there's a stated preference. I don't really care, so if you think it would be better that way, I'll change it.

Reply:
Yeah, we don't have a concrete coding style.  If it's not too much trouble, go ahead and try that style.